### PR TITLE
Type DateTime parameters by default as DateTime2, and use NULL literals

### PR DESCRIPTION
**Pull request by EamonNerbonne**
This is definitely a breaking change, but we want this so that the optimistic concurrency check is possible for datetime2 columns

However, it will break optimistic concurrency checks - and possibly other equality checks - for datetime columns, but at least it's *possible* to work around that issue since there's no data-loss involved

### CODE REVIEW

File: src/ProgressOnderwijsUtils/Data/QueryScalarParameterComponent.cs [hunk 1]
```diff
@@ -15,8 +15,12 @@ public void ToSqlParameter(ref SqlParamArgs paramArgs)
         public static void AppendScalarParameter<TCommandFactory>(ref TCommandFactory factory, [CanBeNull] object o)
             where TCommandFactory : struct, ICommandFactory
         {
-            var param = new QueryScalarParameterComponent { EquatableValue = o ?? DBNull.Value };
-            ParameterizedSqlFactory.AppendSql(ref factory, factory.RegisterParameterAndGetName(param));
+            if (o == null || o == DBNull.Value) {
```
Comment (EamonNerbonne) [inline src/ProgressOnderwijsUtils/Data/QueryScalarParameterComponent.cs]: Previously, I did this in generic edit, but it's probably better to just do it for all queries.

Conceivably this too might cause queries to break.

However, running our test set locally doesn't show anything dramatic, in any case.

**Comment by markijbema** [comment]
dit zou oa dit moeten fixen: https://uocg.manuscript.com/f/cases/31251/foutmelding-Machtiging-batch-verwerkt

**Comment by EamonNerbonne** [comment]
oh je oh je - op staging... good luck :-)

**Comment by markijbema** [comment]
het alternatief was dat edits met datetimes niet werkten

**Comment by EamonNerbonne** [comment]
Oftewel test coverage is niet toerijkend

**Comment by markijbema** [comment]
apparantly, maar dat is niet heel gek; we nemen aan dat die businessedits overal en nergens getest worden, maar nu zijn we begonnen ze om te zetten en begonnen met de minst centrale tabellen, dus die worden juist het minst geraakt

**Comment by EamonNerbonne** [comment]
Ja, en niet elk gebruik van een problematische edit hoeft de bug te triggeren.

**Pull request merged**